ta/pkcs11/src/processing.c

```diff
@@ -538,13 +534,38 @@ enum pkcs11_rc entry_processing_key(struct pkcs11_client *client,
 	struct pkcs11_object *parent = NULL;
 	struct obj_attrs *head = NULL;
 	size_t template_size = 0;
+	void *in_buf = NULL;
+	uint32_t in_size = 0;
 	void *out_buf = NULL;
 	uint32_t out_size = 0;
+	enum processing_func operation = PKCS11_FUNCTION_ENCRYPT;
 
-	if (!client || ptypes != exp_pt ||
+	if (!client ||
+	    TEE_PARAM_TYPE_GET(ptypes, 0) != TEE_PARAM_TYPE_MEMREF_INOUT ||
+	    TEE_PARAM_TYPE_GET(ptypes, 2) != TEE_PARAM_TYPE_MEMREF_OUTPUT ||
 	    out->memref.size != sizeof(obj_handle))
 		return PKCS11_CKR_ARGUMENTS_BAD;
```

Comment:
I would prefer params '1 and '3 also checked. Maybe
```patch
 	if (!client ||
 	    TEE_PARAM_TYPE_GET(ptypes, 0) != TEE_PARAM_TYPE_MEMREF_INOUT ||
 	    TEE_PARAM_TYPE_GET(ptypes, 2) != TEE_PARAM_TYPE_MEMREF_OUTPUT ||
 	    out->memref.size != sizeof(obj_handle) ||
+	    TEE_PARAM_TYPE_GET(ptypes, 3) != TEE_PARAM_TYPE_NONE)
 		return PKCS11_CKR_ARGUMENTS_BAD;
 
 	switch (function) {
 	case PKCS11_FUNCTION_UNWRAP:
 		(...)
		break;
 	case PKCS11_FUNCTION_DERIVE:
+		if (TEE_PARAM_TYPE_GET(ptypes, 1) != TEE_PARAM_TYPE_NONE)
+			return PKCS11_CKR_ARGUMENTS_BAD;
+
 		operation = PKCS11_FUNCTION_ENCRYPT;
 		break;
 	default:
 		return PKCS11_CKR_ARGUMENTS_BAD;
```

Reply:
Ok